cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2055,22 +2055,14 @@ endmacro()
 
 if(ARROW_WITH_GRPC)
   if(c-ares_SOURCE STREQUAL "AUTO")
-    find_package(c-ares QUIET)
+    find_package(c-ares QUIET CONFIG)
     if(NOT c-ares_FOUND)
-      # Fedora doesn't package the CMake config
```

Comment:
Does this mean building anything which requires gRPC on Fedora will fail?

Reply:
No -- it means that the system package cannot be used to build gRPC from source. So if `-DgRPC_SOURCE=BUNDLED` and the system c-ares is found, then gRPC will fail to build